byron/ledger/executable-spec/src/Ledger/Update/Generators.hs

```diff
@@ -0,0 +1,58 @@
+-- |
+
+module Ledger.Update.Generators where
+
+import Hedgehog (Gen)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import Ledger.Core (SlotCount (SlotCount), unBlockCount)
+import Ledger.GlobalParams (k)
+import Ledger.Update (PParams (PParams))
+
+-- | Protocol parameters generator.
+--
+-- TODO: The protocol parameters still needs to be aligned with the formal
+-- spec.
+pps :: Gen PParams
+pps
+  = PParams
+  <$> gMaxHeaderSize
+  <*> gMaxBlockSize
+  <*> gMaxTxSize
+  <*> gMaxProposalSize
+  <*> gBlockSigCntThreshold
+  <*> gSlotsPerEpoch
+  <*> gUpdateProposalTTL
+  <*> pure 1 -- _scriptVersion
+  <*> gConfirmationTreshold
+  <*> gAdoptinThreshold
+  <*> pure k -- _stableAfter
+  <*> pure 0 -- factor @a@
+  <*> pure 0 -- factor @b@
+  where
+    -- In mainet the maximum header size is set to 2000000 and the maximum
+    -- block size is also set to 2000000, so we have to make sure we cover
+    -- those values here. The upper bound is arbitrary though.
+    --
+    -- TODO: For a more realistic lower bound we should figure out the
+    -- minimum block size.
+    gMaxHeaderSize = Gen.integral (Range.constant 0 4000000)
```

Comment:
I don't think 0 is a good max size. If we want to test 0 though, which might make sense , then an equal distribution between 0 and 40000000 will almost never generate 0, maybe better make this a choice between `[0,1]` and `[2,4000000]` ?

Reply:
Good point. I changed this to:

```haskell
    gMaxHeaderSize = Gen.choice
                     [ pure 0
                     , pure 1
                     , Gen.integral (Range.constant 2 4000000)
                     ]
```

Reply:
I think its better like that (but maybe check whether now too many unusable traces are generated. Not sure what the best combination would be.

Reply:
Good point! That might be a problem indeed. I might have to add a frequency instead.